crates/node/src/lib.rs

```diff
@@ -674,6 +674,35 @@ async fn write_initial_value_dont_fail_if_it_exists<T: Versioned + StorageEncode
     }
 }
 
+async fn write_initial_logs_dont_fail_if_it_exists(
+    metadata_store_client: &MetadataStoreClient,
+    key: ByteString,
+    initial_value: Logs,
+) -> Result<(), ReadWriteError> {
+    let value = metadata_store_client
+        .read_modify_write(key, |current| match current {
+            None => Ok(initial_value.clone()),
+            Some(logs) => {
+                if logs.version() == Version::MIN && logs.num_logs() == 0 {
+                    let builder = initial_value.clone().into_builder();
+                    // make sure version is incremented to MIN + 1
+                    Ok(builder.build())
+                } else {
+                    Err(ReadModifyWriteError::FailedOperation(AlreadyInitialized))
+                }
+            }
+        })
+        .await;
```

Comment:
The check for `num_logs()` is a bit extra but in general it looks correct. I would suggest to improve this a bit by checking if the `current_value` == the `initial_value` in case we were trying to overwrite our own previous write. In that case we can just return success instead of bumping up the version.

Reply:
Sounds good!

Reply:
Done, but because Logs does not implement PartitalEq, I compared the config instead since this is what we care about. Instead of adding PartialEq everywhere.